src/app/public/modules/progress-indicator/progress-indicator.component.ts

```diff
@@ -114,8 +120,12 @@ export class SkyProgressIndicatorComponent implements AfterContentInit, OnDestro
         firstItem.isActive = true;
       }
     }
-    this.progressChanges.emit({
-      activeIndex: this.activeIndex
+
+    // setTimeout required to ensure listeners aren't triggered until the view lifecycle has completed. Issue: blackbaud/skyux2#2221
```

Comment:
Maybe something like this?

```
// A timeout is needed to ensure that subscriptions are executed after all lifecycle hooks have completed.
// For example, if a subscriber executed `detectChanges`, it would short-circuit any remaining lifecycle hooks.
```